src/coreclr/jit/lsrabuild.cpp

```diff
@@ -1978,6 +1988,20 @@ void LinearScan::buildPhysRegRecords()
         RegRecord* curr = &physRegs[reg];
         curr->regOrder  = (unsigned char)i;
     }
+
+#if defined(TARGET_XARCH)
+    // xarch has mask registers available when EVEX is supported
+
+    if (compiler->canUseEvexEncoding())
+    {
+        for (unsigned int i = 0; i < lsraRegOrderMskSize; i++)
+        {
+            regNumber  reg  = lsraRegOrderMsk[i];
+            RegRecord* curr = &physRegs[reg];
+            curr->regOrder  = (unsigned char)i;
+        }
+    }
+#endif // TARGET_XARCH
```

Comment:
Much of the TP regression comes from this. LSRA is very sensitive to the additional cost from building these extra 8 registers.